src/request-pipeline/destination-request/index.js

```diff
@@ -33,41 +32,60 @@ export default class DestinationRequest extends EventEmitter {
         this._send();
     }
 
-    _send () {
+    _send (waitData) {
```

Comment:
whatever, `waitForData` is an unclear name for me.
It is a boolean flag.
So, I propose something like `shouldWaitOnDataEvent`.